### Задание №1.
Даны значения величины заработной платы заемщиков банка (zp) и значения их
поведенческого кредитного скоринга (ks):

In [1]:
import numpy as np

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны.    
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

__Решение:__

Ковариацию двух наших величин найдём по формуле несмещённой оценки:

$$\normalsize \sigma_{XY} = \frac{1}{n-1}\sum_{i=1}^{n}\big(x_i-\overline{X}\big)\cdot\big(y_i-\overline{Y}\big),$$

где $\overline{X}, \overline{Y}$ - средние выборочные величин $X, Y$.

In [3]:
m_zp = zp.sum() / zp.shape[0]
m_ks = ks.sum() / ks.shape[0]

cov_zp_ks = ((zp - m_zp) * (ks - m_ks)).sum() / (zp.shape[0] - 1)
cov_zp_ks

10175.37777777778

Проверим полученное значение ковариации функцией `cov()` библиотеки `numpy`: 

In [4]:
np.cov(zp, ks, ddof=1)[0, 1]

10175.377777777776

Коэффициент корреляции Пирсона найдем по формуле:

$$\normalsize r_{XY} = \frac{\sigma_{XY}}{\sigma_X\cdot\sigma_Y},$$

где $\sigma_{XY}$ - ковариация двух величин X, Y, $\,\,\,\sigma_X, \sigma_Y$ - среднеквадратичные отклонения величин X, Y.

Среднеквадратичные отклонения найдём по формуле:

$$\normalsize \sigma_X = \sqrt{\frac{1}{n-1}\sum_{i=1}^n\Big(x_i-\overline{X}\Big)^2},$$

где $\overline{X}$ - среднее выборочное.

In [5]:
std_zp = np.sqrt(((zp - m_zp) ** 2).sum() / (zp.shape[0] - 1))
std_ks = np.sqrt(((ks - m_ks) ** 2).sum() / (ks.shape[0] - 1))

corr_zp_ks = cov_zp_ks / (std_zp * std_ks)
corr_zp_ks

0.8874900920739164

Проверим полученное значение коэффициента корреляции Пирсона функцией `corrcoef()` библиотеки `numpy`:

In [6]:
np.corrcoef(zp, ks)[0, 1]

0.8874900920739162

__Ответ:__

$\normalsize \sigma_{XY} \approx 10175.38\,\,\,\,\, r_{XY} \approx 0.88749$
___

### Задание №2.
Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

In [7]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью $p=0.95$.

__Решение:__

Так как по условию нам известно, что генеральная совокупность имеет нормальное распределение и $\alpha = 1 - p = 0.05$, и неизвестно среднеквадратичное отклонение, то для вычисления ширины доверительного интервала математического ожидания будем использовать t-статистику и её распределение Стьюдента.

$$\normalsize t = \frac{\overline{X}-\mu}{\sigma_X/\sqrt{n}}$$

Границы доверительного интервала математического ожидания найдем из двойного неравенства:

$$\normalsize t_{\alpha/2} \le t \le t_{1-\alpha/2}$$

$$\normalsize t_{\alpha/2} \le \frac{\overline{X}-\mu}{\sigma_X/\sqrt{n}} \le t_{1-\alpha/2}$$

$$\normalsize \frac{t_{\alpha/2}\cdot\sigma_X}{\sqrt{n}} \le \overline{X}-\mu \le \frac{t_{1-\alpha/2}\cdot\sigma_X}{\sqrt{n}}$$

$$\normalsize \overline{X}-\frac{t_{1-\alpha/2}\cdot\sigma_X}{\sqrt{n}} \le \mu \le \overline{X}-\frac{t_{\alpha/2}\cdot\sigma_X}{\sqrt{n}}$$

Найдем недостающие квантили порядка $\alpha/2$ и $1-\alpha/2$ через библиотеку `scipy` и сразу посчитаем границы интервала:

In [8]:
m_iq = iq.mean()
std_iq = iq.std()
n = iq.shape[0]
alpha = 0.05

In [9]:
from scipy import stats

t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-2.262157162740992, 2.2621571627409915)

In [10]:
m_iq - t2 * std_iq / np.sqrt(n), m_iq - t1 * std_iq / np.sqrt(n)

(110.9432125583706, 125.25678744162938)

__Ответ:__

С вероятностью $\normalsize p=0.95,\,\,\,\mu \in \big(110.94, 125.26\big).$
___

### Задание №3.
Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

__Решение:__

$\normalsize \sigma_X = \sqrt{25} = 5,\,\,\,n=27,\,\,\,\overline{X}=174.2,\,\,\,\alpha = 1-p = 0.05$.

По аналогичным формулам t-статистики и ее распределения Стьюдента решим эту задачу.

In [11]:
m_h = 174.2
n = 27
std_h = 5

In [12]:
t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-2.0555294386428713, 2.055529438642871)

In [13]:
m_h - t2 * std_h / np.sqrt(n), m_h - t1 * std_h / np.sqrt(n)

(172.2220658754539, 176.17793412454608)

__Ответ:__

С вероятностью $\normalsize p=0.95,\,\,\,\mu \in \big(172.22, 176.18\big).$